addons/purchase/purchase.py

```diff
@@ -584,6 +584,7 @@ def onchange_product_id(self):
 
         fpos = self.order_id.fiscal_position_id
         self.taxes_id = fpos.map_tax(self.product_id.supplier_taxes_id)
+        self.price_unit = self.env['account.tax']._fix_tax_included_price(self.price_unit, self.product_id.supplier_taxes_id, self.taxes_id)
```

Comment:
You should replace:
`price_unit = seller.price if seller else 0.0`
With:
`price_unit = self.env['account.tax']._fix_tax_included_price(seller.price, self.product_id.supplier_taxes_id, self.taxes_id) if seller else 0.0`

This appears 4 times in purchase.py